nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/NviCandidateRepository.java

```diff
@@ -2,93 +2,161 @@
 import software.amazon.awssdk.enhanced.dynamodb.model.TransactWriteItemsEnhancedRequest;
 import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
 
-public class NviCandidateRepository extends DynamoRepository  {
+public class NviCandidateRepository extends DynamoRepository {
+
     private final DynamoDbTable<CandidateDao> candidateTable;
     private final DynamoDbTable<CandidateUniquenessEntry> uniquenessTable;
     private final DynamoDbIndex<CandidateDao> publicationIdIndex;
+    private final DynamoDbTable<ApprovalStatusDao> approvalStatusTable;
 
     public NviCandidateRepository(DynamoDbClient client) {
         super(client);
-        this.candidateTable = this.client.table(NVI_TABLE_NAME, CandidateDao.TABLE_SCHEMA);
-        this.uniquenessTable = this.client.table(NVI_TABLE_NAME, CandidateUniquenessEntry.TABLE_SCHEMA);
+        this.candidateTable = this.client.table(NVI_TABLE_NAME, fromImmutableClass(CandidateDao.class));
+        this.uniquenessTable = this.client.table(NVI_TABLE_NAME, fromImmutableClass(CandidateUniquenessEntry.class));
         this.publicationIdIndex = this.candidateTable.index(SECONDARY_INDEX_PUBLICATION_ID);
+        this.approvalStatusTable = this.client.table(NVI_TABLE_NAME, fromImmutableClass(ApprovalStatusDao.class));
     }
 
+    public Candidate create(DbCandidate dbCandidate, List<DbApprovalStatus> approvalStatuses) {
+        var identifier = UUID.randomUUID();
+        var candidate = new CandidateDao(identifier, dbCandidate);
+        var uniqueness = new CandidateUniquenessEntry(dbCandidate.publicationId().toString());
+        var transactionBuilder = TransactWriteItemsEnhancedRequest.builder();
+
+        transactionBuilder.addPutItem(this.candidateTable, insertTransaction(candidate, CandidateDao.class));
+
+        approvalStatuses
+            .stream().map(as -> new ApprovalStatusDao(identifier, as))
+            .forEach(as -> transactionBuilder.addPutItem(this.approvalStatusTable,
+                                                         insertTransaction(as,
+                                                                           ApprovalStatusDao.class)));
+        transactionBuilder.addPutItem(this.uniquenessTable,
+                                      insertTransaction(uniqueness, CandidateUniquenessEntry.class));
+
+        this.client.transactWriteItems(transactionBuilder.build());
+        var candidateObj = candidateTable.getItem(candidate);
+        var approvalStatusList = approvalStatusTable.query(
+                queryCandidateParts(identifier, ApprovalStatusDao.TYPE))
+                                     .items()
+                                     .stream()
+                                     .map(ApprovalStatusDao::approvalStatus)
+                                     .toList();
+
+        return new Candidate(identifier, candidateObj.candidate(), approvalStatusList);
+    }
```

Comment:
I think this method could be refactored a bit

Reply:
It looks like this method should be private

Reply:
It's used in NviService. I will refactor